pkg/snapshot/controllers/snapshotschedule.go

```diff
@@ -246,6 +263,54 @@ func (s *SnapshotScheduleController) updateVolumeSnapshotStatus(snapshotSchedule
 	return nil
 }
 
+// cleanupErroredSnapshots cleans any volumesnapshot CRs that are in Error state. It doesn't remove them
+// from the volumesnapshotschedule items list though.
+func (s *SnapshotScheduleController) cleanupErroredSnapshots(snapshotSchedule *stork_api.VolumeSnapshotSchedule) (err error) {
+	for policy, policyVolumeSnapshot := range snapshotSchedule.Status.Items {
+		for idx, snapshot := range policyVolumeSnapshot {
+			// Move to the next snapshot if the current one doesn't meet criteria to be deleted.
+			if snapshot.Deleted ||
+				snapshot.Status != snapv1.VolumeSnapshotConditionError ||
+				time.Since(snapshot.CreationTimestamp.Time) < errorSnapshotCleanupCutoffTime {
+				continue
+			}
+
+			// Filter out the errored out snapshots that are older than the cleanup cutoff period.
+			// Fetch the latest status as well in order to mitigate any late volumesnapshot updates.
+			snapshotStatus, snapshotError, err := getVolumeSnapshotStatus(snapshot.Name, snapshotSchedule.Namespace)
+			if err != nil {
+				log.VolumeSnapshotScheduleLog(snapshotSchedule).Warnf("Error getting status of snapshot %s: %s", snapshot.Name, err.Error())
+				continue
+			}
+
+			if snapshotStatus == snapv1.VolumeSnapshotConditionError {
+				log.VolumeSnapshotScheduleLog(snapshotSchedule).Infof("Going to delete the errored out snapshot: %v,createdAt: %v,error: %v", snapshot.Name, snapshot.CreationTimestamp.Time, snapshotError)
+				err := k8sextops.Instance().DeleteSnapshot(snapshot.Name, snapshotSchedule.Namespace)
+				if err != nil && !errors.IsNotFound(err) {
+					log.VolumeSnapshotScheduleLog(snapshotSchedule).Warnf("Error deleting errored out snapshot %v: %v", snapshot.Name, err)
+					continue
+				}
```

Comment:
nit.. We should try to aim for complete coverage unless something is actually blocking us. If "Added lines #L290 - L291 were not covered by tests" messages come up then there has to be a reason for ignoring these.

Reply:
@pp511 Got it. I'll add tests for these scenarios as well.

Reply:
But ideally, in these cases codecov should fail as well, right? Most of the times, we don't check the `Files changed` tab of a PR where these messages come up.